### PR TITLE
Always start marshal server even if micro-batch is disabled

### PR DESCRIPTION
**Pull request by ssheng**
## Description

Marshal server is always started at serving time. If micro-batching is enabled, marshal server will perform batching; otherwise, requests to the marshal server are pass through. Refactored `start_dev_server` to follow the same pattern as `start_prod_server`.

## Motivation and Context

Marshal server will start functioning like a sidecar or reverse proxy. The change enables further functionality to be added to the marshal server.

## How Has This Been Tested?

`./ci/unit_tests.py` but I'm still investigating the right fix to an existing test. 

## Types of changes

- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature and improvements (non-breaking change which adds/improves functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Code Refactoring (internal change which is not user facing)
- [x] Documentation
- [x] Test, CI, or build

## Component(s) if applicable
- [ ] BentoService (service definition, dependency management, API input/output adapters)
- [ ] Model Artifact (model serialization, multi-framework support)
- [x] Model Server (mico-batching, dockerisation, logging, OpenAPI, instruments)
- [ ] YataiService gRPC server (model registry, cloud deployment automation)
- [ ] YataiService web server (nodejs HTTP server and web UI)
- [ ] Internal (BentoML's own configuration, logging, utility, exception handling)
- [x] BentoML CLI

## Checklist:

- [ ] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly

### CODE REVIEW

File: tests/cli/test_cli.py [hunk 1]
```diff
@@ -207,6 +207,7 @@ def test_echo_docker_api_result_push():
 
 
 @pytest.mark.skipif('not psutil.POSIX')
+@pytest.mark.skip(reason="I'm investigating the right fix for this test.")
```
Comment (ssheng) [inline tests/cli/test_cli.py:210]: Will fix.

File: tests/utils/test_usage_stats.py [hunk 2]
```diff
@@ -117,6 +118,7 @@ def test_track_cli_with_keyboard_interrupt(bento_bundle_path):
             cli = create_bentoml_cli()
             runner.invoke(cli.commands['serve'], [bento_bundle_path])
             _, properties = track.call_args_list[0][0]
+            print(properties)
```
Comment (ssheng) [inline tests/utils/test_usage_stats.py:121]: Will remove.

**Comment by bojiang** [comment]
We can only enable the batching layer by default without removing the `--enable-microbatch` option.
`enable_microbatch` could still be False, then the batching layer would just act like a reverse proxy. Hooking `setup_routes_from_pb` in the bentoml/marshal package will do the trick.

**Comment by ssheng** [comment]
@bojiang thanks for the explanation. Now more dots of our discussion start to connect. Please see if the new commit addresses you concerns.

File: bentoml/marshal/marshal.py [hunk 3]
```diff
@@ -360,10 +368,14 @@ def make_app(self):
...
+    def fork_start_app(
+        self, port=Provide[BentoMLContainer.config.api_server.port],
+    ):
         # Use new eventloop in the fork process to avoid problems on MacOS
         # ref: https://groups.google.com/forum/#!topic/python-tornado/DkXjSNPCzsI
         loop = asyncio.new_event_loop()
         asyncio.set_event_loop(loop)
         app = self.make_app()
+
+        logger.info("Running micro batch service on :%d", port)
```
Comment (ssheng) [inline bentoml/marshal/marshal.py:380]: Will remove.

**Review by bojiang — APPROVED** [review]

**Pull request merged**